### DISCUSSION TITLE
Unable to run Wii U emulator (Vulkan issue)

### DISCUSSION THREAD

**Discussion by Jue-Xu** [The-Wineskin-Project/WineskinServer#236] [Q&A] [answered]
I am trying to run a Wii U emulator (http://cemu.info/) through Wineskin on my M1 Macbook air. The cemu.exe can run, but when I start the game it crashed with the error
`error when initializing Vulkan renderer: unable to create a logical device: -8`

I have tried running `dxvk_macos.verb` in winetricks. Both wrapper and engine are lastest. I also tried copy the libMoltenVK.dylib to the lib64 folder as this video did (https://www.youtube.com/watch?v=Y7e5cU7-d9c), but it doesn't work. 

I am not sure how to fix it to reach the screen in https://www.reddit.com/r/cemu/comments/oznk8y/managed_to_run_cemu_botw_in_my_m1_mac_finally/. 
Any help is appreciated.

The LastRunWine.log:

> 00f4:fixme:kernelbase:AppPolicyGetThreadInitializationType FFFFFFFFFFFFFFFA, 00000000011FFE10
> 0024:fixme:wbemprox:client_security_SetBlanket 00000002419BED50, 0000000000A57EA0, 10, 0, (null), 3, 3, 0000000000000000, 0x00000000
> 0024:fixme:wbemprox:client_security_Release 00000002419BED50
> 0024:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
> 00fc:fixme:seh:get_thread_times not implemented on this platform
> 00fc:fixme:service:I_ScRegisterDeviceNotification Notification filters are not yet implemented.
> 00fc:fixme:service:I_ScRegisterDeviceNotification Notification filters are not yet implemented.
> 00fc:fixme:rawinput:RegisterRawInputDevices Unhandled flags 0x2100 for device 0.
> 00fc:fixme:setupapi:CM_Get_Parent 0000000001A3F414 0x00000000 0x00000000 stub
> 00fc:fixme:setupapi:CM_Get_Parent 0000000001A3F414 0x00000000 0x00000000 stub
> 00fc:fixme:setupapi:CM_Get_Parent 0000000001A3F414 0x00000000 0x00000000 stub
> 00fc:fixme:setupapi:CM_Get_Parent 0000000001A3F414 0x00000000 0x00000000 stub
> 00fc:fixme:msg:ChangeWindowMessageFilter c051 00000001
> 00fc:fixme:msg:ChangeWindowMessageFilter 111 00000001
> 00bc:fixme:imm:ImeSetActiveContext (0xb71e0, 0): stub
> 00bc:fixme:imm:ImmReleaseContext (0000000000010022, 00000000000B71E0): stub
> 00fc:fixme:imm:ImeSetActiveContext (0x18051a0, 1): stub
> 00fc:fixme:imm:ImmReleaseContext (0000000000010082, 00000000018051A0): stub
> [mvk-info] MoltenVK version 1.1.3, supporting Vulkan version 1.1.170.
> 	The following 71 Vulkan extensions are supported:
> 		VK_KHR_16bit_storage v1
> 		VK_KHR_8bit_storage v1
> 		VK_KHR_bind_memory2 v1
> 		VK_KHR_create_renderpass2 v1
> 		VK_KHR_dedicated_allocation v3
> 		VK_KHR_depth_stencil_resolve v1
> 		VK_KHR_descriptor_update_template v1
> 		VK_KHR_device_group v4
> 		VK_KHR_device_group_creation v1
> 		VK_KHR_driver_properties v1
> 		VK_KHR_external_fence v1
> 		VK_KHR_external_fence_capabilities v1
> 		VK_KHR_external_memory v1
> 		VK_KHR_external_memory_capabilities v1
> 		VK_KHR_external_semaphore v1
> 		VK_KHR_external_semaphore_capabilities v1
> 		VK_KHR_get_memory_requirements2 v1
> 		VK_KHR_get_physical_device_properties2 v2
> 		VK_KHR_get_surface_capabilities2 v1
> 		VK_KHR_image_format_list v1
> 		VK_KHR_maintenance1 v2
> 		VK_KHR_maintenance2 v1
> 		VK_KHR_maintenance3 v1
> 		VK_KHR_multiview v1
> 		VK_KHR_portability_subset v1
> 		VK_KHR_push_descriptor v2
> 		VK_KHR_relaxed_block_layout v1
> 		VK_KHR_sampler_mirror_clamp_to_edge v3
> 		VK_KHR_sampler_ycbcr_conversion v14
> 		VK_KHR_shader_draw_parameters v1
> 		VK_KHR_shader_float16_int8 v1
> 		VK_KHR_shader_subgroup_extended_types v1
> 		VK_KHR_storage_buffer_storage_class v1
> 		VK_KHR_surface v25
> 		VK_KHR_swapchain v70
> 		VK_KHR_swapchain_mutable_format v1
> 		VK_KHR_timeline_semaphore v2
> 		VK_KHR_uniform_buffer_standard_layout v1
> 		VK_KHR_variable_pointers v1
> 		VK_EXT_debug_marker v4
> 		VK_EXT_debug_report v9
> 		VK_EXT_debug_utils v2
> 		VK_EXT_descriptor_indexing v2
> 		VK_EXT_fragment_shader_interlock v1
> 		VK_EXT_hdr_metadata v2
> 		VK_EXT_host_query_reset v1
> 		VK_EXT_image_robustness v1
> 		VK_EXT_inline_uniform_block v1
> 		VK_EXT_memory_budget v1
> 		VK_EXT_metal_surface v1
> 		VK_EXT_post_depth_coverage v1
> 		VK_EXT_private_data v1
> 		VK_EXT_robustness2 v1
> 		VK_EXT_scalar_block_layout v1
> 		VK_EXT_shader_stencil_export v1
> 		VK_EXT_shader_viewport_index_layer v1
> 		VK_EXT_subgroup_size_control v2
> 		VK_EXT_swapchain_colorspace v4
> 		VK_EXT_texel_buffer_alignment v1
> 		VK_EXT_texture_compression_astc_hdr v1
> 		VK_EXT_vertex_attribute_divisor v3
> 		VK_AMD_gpu_shader_half_float v2
> 		VK_AMD_negative_viewport_height v1
> 		VK_AMD_shader_image_load_store_lod v1
> 		VK_AMD_shader_trinary_minmax v1
> 		VK_IMG_format_pvrtc v1
> 		VK_INTEL_shader_integer_functions2 v1
> 		VK_GOOGLE_display_timing v1
> 		VK_MVK_macos_surface v3
> 		VK_MVK_moltenvk v31
> 		VK_NV_glsl_shader v1
> [mvk-info] GPU device:
> 		model: Apple M1
> 		type: Integrated
> 		vendorID: 0x106b
> 		deviceID: 0xa120
> 		pipelineCacheUUID: 00002777-0400-03ED-0000-000000000000
> 	supports the following Metal Versions, GPU's and Feature Sets:
> 		Metal Shading Language 2.2
> 		GPU Family Apple 5
> 		GPU Family Apple 4
> 		GPU Family Apple 3
> 		GPU Family Apple 2
> 		GPU Family Apple 1
> 		GPU Family Mac 2
> 		GPU Family Mac 1
> 		GPU Family Common 3
> 		GPU Family Common 2
> 		GPU Family Common 1
> 		macOS GPU Family 2 v1
> 		macOS GPU Family 1 v4
> 		macOS GPU Family 1 v3
> 		macOS GPU Family 1 v2
> 		macOS GPU Family 1 v1
> [mvk-info] Created VkInstance for Vulkan version 1.1.170, as requested by app, with the following 3 Vulkan extensions enabled:
> 		VK_KHR_surface v25
> 		VK_EXT_debug_utils v2
> 		VK_EXT_metal_surface v1
> info:  Game: Cemu.exe
> info:  DXVK: v1.7-14-gef993b2f
> info:  Built-in extension providers:
> info:    Win32 WSI
> info:    OpenVR
> warn:  OpenVR: Failed to locate module
> info:  Enabled instance extensions:
> info:    VK_KHR_get_surface_capabilities2
> info:    VK_KHR_surface
> info:    VK_KHR_win32_surface
> [mvk-info] MoltenVK version 1.1.3, supporting Vulkan version 1.1.170.
> 	The following 71 Vulkan extensions are supported:
> 		VK_KHR_16bit_storage v1
> 		VK_KHR_8bit_storage v1
> 		VK_KHR_bind_memory2 v1
> 		VK_KHR_create_renderpass2 v1
> 		VK_KHR_dedicated_allocation v3
> 		VK_KHR_depth_stencil_resolve v1
> 		VK_KHR_descriptor_update_template v1
> 		VK_KHR_device_group v4
> 		VK_KHR_device_group_creation v1
> 		VK_KHR_driver_properties v1
> 		VK_KHR_external_fence v1
> 		VK_KHR_external_fence_capabilities v1
> 		VK_KHR_external_memory v1
> 		VK_KHR_external_memory_capabilities v1
> 		VK_KHR_external_semaphore v1
> 		VK_KHR_external_semaphore_capabilities v1
> 		VK_KHR_get_memory_requirements2 v1
> 		VK_KHR_get_physical_device_properties2 v2
> 		VK_KHR_get_surface_capabilities2 v1
> 		VK_KHR_image_format_list v1
> 		VK_KHR_maintenance1 v2
> 		VK_KHR_maintenance2 v1
> 		VK_KHR_maintenance3 v1
> 		VK_KHR_multiview v1
> 		VK_KHR_portability_subset v1
> 		VK_KHR_push_descriptor v2
> 		VK_KHR_relaxed_block_layout v1
> 		VK_KHR_sampler_mirror_clamp_to_edge v3
> 		VK_KHR_sampler_ycbcr_conversion v14
> 		VK_KHR_shader_draw_parameters v1
> 		VK_KHR_shader_float16_int8 v1
> 		VK_KHR_shader_subgroup_extended_types v1
> 		VK_KHR_storage_buffer_storage_class v1
> 		VK_KHR_surface v25
> 		VK_KHR_swapchain v70
> 		VK_KHR_swapchain_mutable_format v1
> 		VK_KHR_timeline_semaphore v2
> 		VK_KHR_uniform_buffer_standard_layout v1
> 		VK_KHR_variable_pointers v1
> 		VK_EXT_debug_marker v4
> 		VK_EXT_debug_report v9
> 		VK_EXT_debug_utils v2
> 		VK_EXT_descriptor_indexing v2
> 		VK_EXT_fragment_shader_interlock v1
> 		VK_EXT_hdr_metadata v2
> 		VK_EXT_host_query_reset v1
> 		VK_EXT_image_robustness v1
> 		VK_EXT_inline_uniform_block v1
> 		VK_EXT_memory_budget v1
> 		VK_EXT_metal_surface v1
> 		VK_EXT_post_depth_coverage v1
> 		VK_EXT_private_data v1
> 		VK_EXT_robustness2 v1
> 		VK_EXT_scalar_block_layout v1
> 		VK_EXT_shader_stencil_export v1
> 		VK_EXT_shader_viewport_index_layer v1
> 		VK_EXT_subgroup_size_control v2
> 		VK_EXT_swapchain_colorspace v4
> 		VK_EXT_texel_buffer_alignment v1
> 		VK_EXT_texture_compression_astc_hdr v1
> 		VK_EXT_vertex_attribute_divisor v3
> 		VK_AMD_gpu_shader_half_float v2
> 		VK_AMD_negative_viewport_height v1
> 		VK_AMD_shader_image_load_store_lod v1
> 		VK_AMD_shader_trinary_minmax v1
> 		VK_IMG_format_pvrtc v1
> 		VK_INTEL_shader_integer_functions2 v1
> 		VK_GOOGLE_display_timing v1
> 		VK_MVK_macos_surface v3
> 		VK_MVK_moltenvk v31
> 		VK_NV_glsl_shader v1
> [mvk-info] GPU device:
> 		model: Apple M1
> 		type: Integrated
> 		vendorID: 0x106b
> 		deviceID: 0xa120
> 		pipelineCacheUUID: 00002777-0400-03ED-0000-000000000000
> 	supports the following Metal Versions, GPU's and Feature Sets:
> 		Metal Shading Language 2.2
> 		GPU Family Apple 5
> 		GPU Family Apple 4
> 		GPU Family Apple 3
> 		GPU Family Apple 2
> 		GPU Family Apple 1
> 		GPU Family Mac 2
> 		GPU Family Mac 1
> 		GPU Family Common 3
> 		GPU Family Common 2
> 		GPU Family Common 1
> 		macOS GPU Family 2 v1
> 		macOS GPU Family 1 v4
> 		macOS GPU Family 1 v3
> 		macOS GPU Family 1 v2
> 		macOS GPU Family 1 v1
> [mvk-info] Created VkInstance for Vulkan version 1.1.0, as requested by app, with the following 3 Vulkan extensions enabled:
> 		VK_KHR_get_surface_capabilities2 v1
> 		VK_KHR_surface v25
> 		VK_EXT_metal_surface v1
> info:  Apple M1:
> info:    Driver: 0.2.1911
> info:    Vulkan: 1.1.170
> info:    Memory Heap[0]: 
> info:      Size: 8192 MiB
> info:      Flags: 0x1
> info:      Memory Type[0]: Property Flags = 0x1
> info:      Memory Type[1]: Property Flags = 0xf
> info:      Memory Type[2]: Property Flags = 0xb
> info:  Adapter LUID 0: 0:3f4
> [mvk-info] Using MTLFence for Vulkan semaphores.
> [mvk-error] VK_ERROR_FEATURE_NOT_PRESENT: vkCreateDevice(): Requested feature is not available on this device.
> [mvk-info] Created VkDevice to run on GPU Apple M1 with the following 3 Vulkan extensions enabled:
> 		VK_KHR_driver_properties v1
> 		VK_KHR_sampler_mirror_clamp_to_edge v3
> 		VK_KHR_swapchain v70
>

**Comment by Gcenx** [comment, marked as answer]
As can be seen in the provided log MoltenVK doesn’t provide the needed Vulkan extensions nothing I can do about this.

The provided MoltenVK-DXVK simply fakes some extensions to make DXVK-macOS happy.
